### PR TITLE
chore!: migrate to cometbft fork

### PR DESCRIPTION
**Pull request by cmwaters**
This ultimately changes the go.mod to using cometbft instead of tendermint which we will be using for the v0.37 release and is what the sdk depends on in v0.47

### CODE REVIEW

File: consensus/state.go [hunk 1]
```diff
@@ -877,7 +877,7 @@ func (cs *State) handleMsg(mi msgInfo) {
 		// We probably don't want to stop the peer here. The vote does not
 		// necessarily comes from a malicious peer but can be just broadcasted by
 		// a typical peer.
-		// https://github.com/tendermint/tendermint/issues/1281
+		// https://github.com/cometbft/cometbft/issues/1281
```
Comment (evan-forbes) [inline consensus/state.go:880]: fwiw I'm don't think its worth going through and finding each broken link as I don't think they're super useful and we can just manually fix the links if we really need to

File: docs/celestia-architecture/adr-004-mvp-light-client.md [hunk 2]
```diff
@@ -259,7 +259,7 @@ A DAS light client will have to request the `DASLightBlock` for this as per abov
 
 Ideally, we add the DAS light client to the existing e2e tests.
 It might be worth to catch up with some relevant changes from tendermint upstream.
-In particular, [tendermint/tendermint#6196](https://github.com/tendermint/tendermint/pull/6196) and previous changes that it depends on.
+In particular, [tendermint/tendermint#6196](https://github.com/cometbft/cometbft/pull/6196) and previous changes that it depends on.
```
Comment (evan-forbes) [inline docs/celestia-architecture/adr-004-mvp-light-client.md:262]: explicit links on the other hand (like http:// whatever) we might actually want to fix

that or disable the dead link linter

**Review by evan-forbes — COMMENTED** [review]
we've changed these to tendermint -> lazyledger-core -> celestia-core -> tendermint -> comet

did not expect to change these so frequently :laughing:

**Review by evan-forbes — APPROVED** [review]

File: go.mod [hunk 3]
```diff
@@ -1,4 +1,4 @@
-module github.com/tendermint/tendermint
+module github.com/cometbft/cometbft
```
Comment (rootulp) [inline go.mod:1]: 👍 

When this PR lands in a release, we should call out in release notes that consumers of this library (i.e. celestia-app and celestia-node) need to change the `replace` statement in go.mod

**Review by rootulp — APPROVED** [review]
Overall LGTM. I scrolled through the diff and most of it seems like fixing the import statements. Agreed w/ Evan on the broken links

**Comment by cmwaters** [comment]
Yeah I'm aware of the broken links and am aware that in doing this quickly I have probably created a few broken links. I don't think this is problematic because I don't think there is anyone using our fork as a reference of some information (other than the ADRs). 

I can use a link checker and go through and correct the places were I mistakenly changed the url in a different PR

**Pull request merged**